### PR TITLE
AK: call_once implementation and usage

### PR DESCRIPTION
**Pull request by ldm5180**
Provide a `call_once` function like `std::call_once` and use it in places which implement the same behavior.

### CODE REVIEW

File: AK/Once.h [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#pragma once
+
+namespace AK {
+
+struct OnceFlag {
+  bool is_previously_called{};
```
[the remark this answers is not in the document]
Comment (ldm5180) [reply AK/Once.h]: My mistake. Fixed.

File: AK/Once.h [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+};
+
+template<class Callable, class... Args>
+constexpr void call_once(OnceFlag& flag, Callable&& f, Args&&... args)
+{
+  if (not flag.is_previously_called) {
```
[the remark this answers is not in the document]
Comment (ldm5180) [reply AK/Once.h]: Done.

File: AK/Base64.cpp [hunk 3]
```diff
@@ -36,15 +37,14 @@ namespace AK {
...
-    if (s_initialized)
-        return;
-    for (size_t i = 0; i < sizeof(s_alphabet) - 1; ++i)
-        s_table[s_alphabet[i]] = i;
-    s_initialized = true;
+  static OnceFlag s_initialized{};
+  call_once(s_initialized, [&] {
+      for (size_t i = 0; i < sizeof(s_alphabet) - 1; ++i)
+          s_table[s_alphabet[i]] = i;
+  });
```
[the remark this answers is not in the document]
Comment (ldm5180) [reply AK/Base64.cpp]: That is true. I should add that!

[the remark this answers is not in the document]
Comment (ldm5180) [reply AK/Base64.cpp]: I don't see any good way to do this without creating a dependency between LibPthread and AK. I am going to close this PR for now.

[the remark this answers is not in the document]
Comment (tomuta) [reply AK/Base64.cpp]: For AK/Base64 I suggest we just use the Singleton class we already have. Which also is thread safe (unlike this implementation).

**Pull request closed without merging**